lib/health_check/utils.rb

```diff
@@ -145,7 +145,15 @@ def self.check_smtp(settings, timeout)
           status = '250'
         else
           smtp = Net::SMTP.new(settings[:address], settings[:port])
-          smtp.enable_starttls if settings[:enable_starttls_auto]
+          openssl_verify_mode = settings[:openssl_verify_mode]
+
+          if openssl_verify_mode.kind_of?(String)
+            openssl_verify_mode = OpenSSL::SSL.const_get("VERIFY_#{openssl_verify_mode.upcase}")
+          end
+
+          ssl_context = Net::SMTP.default_ssl_context
+          ssl_context.verify_mode = openssl_verify_mode if openssl_verify_mode
+          smtp.enable_starttls(ssl_context) if settings[:enable_starttls_auto]
```

Comment:
Here, the test environment has been configured to allow specifying OpenSSL::SSL::VERIFY_NONE to avoid errors caused by SSL certificates generated by fake_smtp_server.rb.